src/components/ProgressLine/index.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useEffect, useState } from 'react';
+import styled from 'styled-components';
+
+type Props = {
+  visualPartition: {
+    percentage: number;
+    color: string;
+  }[];
+};
+
+interface WrapperProps {
+  backgroundColor: string;
+}
+
+interface GraphProps extends WrapperProps {
+  width: number;
+}
+
+const ProgressLine: React.FC<Props> = ({ visualPartition }) => {
+  const resetProgress = visualPartition.map(() => {
+    return 0;
+  });
+
+  const [widths, setWidths] = useState(
+    visualPartition.map(() => {
+      return 0;
+    })
+  );
+
+  useEffect(() => {
+    requestAnimationFrame(() => {
+      setWidths(
+        visualPartition.map((item) => {
+          return item.percentage;
+        })
+      );
+    });
+
+    return () => {
+      resetProgress;
+    };
+  }, [visualPartition]);
+
+  return (
+    <Wrapper backgroundColor={'--color-grey-10'}>
+      {visualPartition.map(({ color }, index) => {
```

Comment:
You don't have to use {} if you directly returning the value from map function
Use this 
`[].map((value) => value)` instead of this
`[].map((value) => {
    return value;
})`